client/lib/signup/step-actions/index.js

```diff
@@ -525,15 +525,10 @@ function addPrivacyProtectionIfSupported( item, state ) {
 export function launchSiteApi( callback, dependencies ) {
 	const { siteSlug } = dependencies;
 
-	wpcom.undocumented().launchSite( siteSlug, function ( error ) {
-		if ( error ) {
-			callback( error );
-
-			return;
-		}
-
-		callback();
-	} );
+	wpcom.req
+		.post( `/sites/${ siteSlug }/launch` )
+		.then( () => callback() )
```

Comment:
Not the ideal way to call the callback function (with no args on success, and error arg on error), but it's already broad throughout the signup framework, so it wouldn't make sense to refactor it here.

Reply:
I changed that to `callback( null )`, although it doesn't make much difference.